generators/_init-hybrid/templates/plain/.circleci/preinstall.js

```diff
@@ -0,0 +1,103 @@
+const fs = require('fs');
+const fsPromises = fs.promises;
+const { sep } = require('path');
+
+const nodeModulesDir = 'node_modules';
+const packageLockFilename = 'package-lock.json';
+const dependenciesProperty = 'dependencies';
+
+/**
+ * Read a JSON file and return the parsed JSON
+ * @param {string} filename Path to file
+ */
+async function readJSONFile(filename) {
+    try {
+        const data = await fsPromises.readFile(filename, 'utf-8');
+        return JSON.parse(data);
+    }
+    catch(err) {
+        console.log(err.message);
+        process.exit(0); // exit without error
+    }
+}
+
+/**
+ * JSON stringify the given object and write it to the given file
+ * @param {string} filename Path to file
+ * @param {any} jsonObj Object structure
+ */
+async function writeJSONFile(filename, jsonObj) {
+    try {
+        fsPromises.writeFile(filename, JSON.stringify(jsonObj, null, 4), 'utf-8');
+    }
+    catch(err) {
+        console.log(err.message);
+        process.exit(0); // exit without error
+    }
+}
+
+/**
+ * Check if the given property exists on the first level of the given object
+ * @param {any} json JSON object
+ * @param {string} property Name of the property
+ */
+function checkPropertyExists(json, property) {
+    if (!json.hasOwnProperty(property)) {
+        console.log(`No ${property} property found in ${packageLockFilename}.`);
+        process.exit(0); // exit without error
+    }
+    return json;
+}
+
+/**
+ * Filter git dependencies by the presence of the `from` property in the given dependency object.
+ * @param {any} json JSON object
+ * @param {string} property Name of the property
+ * @returns Returns a list of package names
+ */
+function findGitDependencies(dependencies) {
+    // filter git dependencies by the presence of the `from` property
+    const gitDependencies = Object.entries(dependencies).filter((entry) => entry[1].from).map((entry) => entry[0]);
```

Comment:
@anita-steiner In my tests only the git dependencies have this `from` property. [The npm documentation does not mention the `from` property at all](https://docs.npmjs.com/files/package-lock.json#dependencies). Alternatively, we could use a regex that searches for github or bitbucket in the [`version` property](https://docs.npmjs.com/files/package-lock.json#version-1). What do you think?